src/controllers/authenticationController.ts

```diff
@@ -21,74 +21,138 @@ export default class AuthenticationController {
     private readonly sessionStorage: ISessionStorage,
     private readonly samlCert: string,
     private readonly spidStrategy: SpidStrategy,
-    private readonly tokenService: TokenService
+    private readonly tokenService: TokenService,
+    private readonly clientProfileRedirectionUrl: string
   ) {}
 
   /**
    * The Assertion consumer service.
    */
   // tslint:disable-next-line:no-any
-  public acs(userPayload: any, res: express.Response): void {
-    winston.log(
-      "info",
-      "Called assertion consumer service with data:",
-      userPayload
-    );
-
+  public async acs(userPayload: any): Promise<Either<Error, IResponse>> {
     const errorOrUser = validateSpidUser(userPayload);
 
-    errorOrUser.fold(
-      (error: Error) => this.return500Error(error, res),
-      (spidUser: SpidUser) => {
-        const user = toAppUser(spidUser, this.tokenService.getNewToken());
+    if (isLeft(errorOrUser)) {
+      const error = errorOrUser.value;
+      return left(error);
+    }
 
-        this.sessionStorage.set(user.token, user);
+    const spidUser = errorOrUser.value;
+    const user = toAppUser(spidUser, this.tokenService.getNewToken());
 
-        const url =
-          process.env.CLIENT_REDIRECTION_URL || "/profile.html?token={token}";
-        const urlWithToken = url.replace("{token}", user.token);
+    const timestamp = Date.now();
+    const errorOrBool = await this.sessionStorage.set(
+      user.token,
+      user,
+      timestamp
+    );
 
-        res.redirect(urlWithToken);
-      }
+    if (isLeft(errorOrBool)) {
+      const error = errorOrBool.value;
+      return left(error);
+    }
+    const response = errorOrBool.value;
+
+    if (!response) {
+      return left(new Error("Error creating the user session"));
+    }
+    const urlWithToken = this.clientProfileRedirectionUrl.replace(
+      "{token}",
```

Comment:
see my comment above on `clientProfileRedirectionUrl`, it could simply become `this. clientProfileRedirectionUrl(user.token)`, thus avoiding splitting the responsibility of replacing the `{token}` string in multiple places